src/components/HomepageManager.jsx

```diff
@@ -1,27 +1,55 @@
-import { getTenArticles } from "../utils/get";
+import { getAllArticles, getAllTopics } from "../utils/get";
 import { useState, useEffect } from "react";
 import ArticleList from "./ArticleList";
 import SortByList from "./SortByList";
 import Grid from "@mui/material/Unstable_Grid2";
 import Loading from "./Loading";
+import { useSearchParams } from "react-router-dom";
 
 export default function HomepageManager() {
   const [articleList, setArticleList] = useState([]);
   const [loading, setLoading] = useState(true);
+  const [topicList, setTopicList] = useState([]);
+  const [topicSearchTerm, setTopicSearchTerm] = useSearchParams();
+  const [sortByTerm, setSortByTerm] = useSearchParams();
+  const sortByList = ["Title", "Author" ]
+
+  const limit = "10";
+  const topic = topicSearchTerm.get("topic");
+  const sort_by = sortByTerm.get("sort_by");
+
+  const setSortTopic = (topic) => {
+    const newTopicTerm = new URLSearchParams(topicSearchTerm);
+    newTopicTerm.set("topic", topic);
+    setTopicSearchTerm(newTopicTerm);
+  };
+
+  const setSortBy = (sort_by) => {
+    const newSortBy = new URLSearchParams(searchTerm);
+    newSortBy.set("sort_by", sort_by);
+    setSortByTerm(newSortBy);
+  };
+
+  useEffect(() => {
+    getAllTopics().then((response) => {
+      setTopicList(response);
+      setLoading(false);
+    });
+  }, [articleList]);
 
   useEffect(() => {
-    getTenArticles().then((response) => {
+    getAllArticles(limit, topic, sort_by).then((response) => {
```

Comment:
Love that you've made this a flexible function which takes in optional arguments 🙌